src/irr/asset/CBAWMeshFileLoader.cpp

```diff
@@ -253,39 +282,69 @@ io::IReadFile* CBAWMeshFileLoader::createConvertIntoVer_spec<3>(SContext& _ctx,
 
         newoffset = offset + offsetDiff;
 
-        if (hdr.blobType == asset::Blob::EBT_FINAL_BONE_HIERARCHY)
+        uint8_t stackmem[1u << 10];
+        uint32_t attempt = 0u;
+        uint8_t decrKey[16];
+        size_t decrKeyLen = 16u;
+
+        void* blob = nullptr;
+
+        /* to state blob's/asset's hierarchy level we'd have to load (and possibly decrypt and decompress) the blob
+         however we don't need to do this here since we know format's version (baw v1) and so we can be sure that hierarchy level for FinalBoneHierarchy is 3 (won't be true for v2 to v3 conversion)
+        */
+
+        auto fetchRawBlob = [&](uint32_t hierarchyLevel)
         {
-            const uint32_t offset = offsets[i];
-
-            uint8_t stackmem[1u << 10];
-            uint32_t attempt = 0u;
-            uint8_t decrKey[16];
-            size_t decrKeyLen = 16u;
-            void* blob = nullptr;
-            /* to state blob's/asset's hierarchy level we'd have to load (and possibly decrypt and decompress) the blob
-            however we don't need to do this here since we know format's version (baw v1) and so we can be sure that hierarchy level for FinalBoneHierarchy is 3 (won't be true for v2 to v3 conversion)
-            */
-            constexpr uint32_t FINALBONEHIERARCHY_HIERARCHY_LVL = 3u;
-            while (_override->getDecryptionKey(decrKey, decrKeyLen, attempt, _baw2file, "", genSubAssetCacheKey(_baw2file->getFileName().c_str(), hdr.handle), _ctx.inner, FINALBONEHIERARCHY_HIERARCHY_LVL))
+            while (_override->getDecryptionKey(decrKey, decrKeyLen, attempt, _baw2file, "", genSubAssetCacheKey(_baw2file->getFileName().c_str(), hdr.handle), _ctx.inner, hierarchyLevel))
             {
                 if (!((hdr.compressionType & asset::Blob::EBCT_AES128_GCM) && decrKeyLen != 16u))
                     blob = tryReadBlobOnStack<asset::legacyv2::BlobHeaderV2>(SBlobData_t<asset::legacyv2::BlobHeaderV2>(&hdr, baseOffsetv2 + offset), _ctx, decrKey, stackmem, sizeof(stackmem));
                 if (blob)
                     break;
                 ++attempt;
             }
+        };
 
-            // patch for meshbuffer blob
-            // TODO
+        // should I actually do anything with meshes if their flipping is dependent on flag placed somewhere in IAssetLoader/Writer?
+        // so it isn't a member of any of blobs
 
-            offsetDiff += static_cast<int32_t>(hdr.blobSizeDecompr) - static_cast<int32_t>(hdr.effectiveSize());
-            hdr.compressionType = asset::Blob::EBCT_RAW;
-            core::XXHash_256(blob, hdr.blobSizeDecompr, hdr.blobHash);
-            hdr.blobSize = hdr.blobSizeDecompr;
+        switch (hdr.blobType)
+        {
+            case asset::Blob::EBT_FINAL_BONE_HIERARCHY:
```

Comment:
there's no reason for you to touch the final bone hierarchy blob because you won't be changing it

Reply:
so should I delete it?

Reply:
yes